src/main/resources/META-INF/resources/primefaces/core/core.dialog.js

```diff
@@ -159,6 +226,10 @@ if (!PrimeFaces.dialog) {
             }
             else {
                 windowContext = rootWindow;
+                var frames = dlgWidget.cfg.sourceFrames;
+                for (var i = 0; i < frames.length; i++) {
+                    windowContext = $(windowContext.document).find(frames[i]).get(0).contentWindow;
```

Comment:
Because there might be multiple nested frames from the top window to the source widget. Think of a frameset containing another frameset. If the source button is in frame1->frame1_1, we need to first resolve form the top window the frame with the id `frame1`, go into the content window of that frame, resolve `frame1_1` and use that content window as windowContext for the source widget.